### ISSUE TITLE
Game displays too large area on tablets

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#43]
```
When playing the game on an Android 7in tablet, the world view shows too much 
of the map. For example, in Crossglen you can see all the way from your house 
down to the smithy.

Displaying that much of the map causes the tiles to be very small, which makes 
it difficult to click as precisely as if the tiles are normal sized.

Instead, the display on tablets should be identical to the display on a smaller 
device, but just upscaled.
```

Original issue reported on code.google.com by `oskar.wi...@gmail.com` on 1 Dec 2010 at 7:09

**Comment by GoogleCodeExporter** [comment]
```
or have it selectable.
For huawei 8110 (I think 240x320) the map is 11x11 tiles (if I'm correct) but I 
have the feeling that everything is scaled down (at least text is somewhat 
blurry)
It's ok for me, but a choice whether or not to scale (or even choose scale) 
would be nice.
```

Original comment by `surrano` on 8 Dec 2010 at 12:18

**Comment by GoogleCodeExporter** [comment]
```
Does this require bigger tiles or are we within the limits that simply blowing 
up the ones in the game now will work and still look good?
```

Original comment by `sdeva...@gmail.com` on 9 Feb 2011 at 11:00

**Comment by GoogleCodeExporter** [comment]
```
At the moment, the game reshapes the draw area to fit the current resolution of 
the phone. I like the fact that it does this, because it allows you to play in 
landscape mode by rotating the phone, allowing you to see further to the left 
or right.

Perhaps people should just have the option to double the sprite size for high 
resolution devices.
```

Original comment by `steve.st...@googlemail.com` on 10 Feb 2011 at 12:26

**Comment by GoogleCodeExporter** [comment]
```
I also like it if this was configurable. But the default should be that 
displaying on a tablet should look the same as on a smaller device. This could 
introduce an advantage to players playing on tablet devices compared to players 
on smaller devices, and that's also something worth considering if we allow 
configuration of this.

There is one issue about resizing the tiles that I haven't been able to figure 
out an elegant solution for. Android has built-in support for resizing images 
in apps to a size that is suitable for the device it is being run on. This 
automatic resizing is done by dithering, to make the images look much better 
when scaled up. The problem though is that the dithering is done on the whole 
image, which is a problem for our tileset images. The effect is that some tiles 
bleed into other tiles, making them look really bad when scaled up by dithering.

The solution would probably be to scale the tiles ourselves, or find some way 
to disable the dithering when using automatic scaling.

In my opinion, this is a high priority issue since the market currently points 
towards a lot more Android tablets, and we should make the game as accessible 
as possible t a wide audience.
```

Original comment by `oskar.wi...@gmail.com` on 10 Feb 2011 at 7:03
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
I think you might be overstating the advantage of being able to see further on 
a high resolution device. I don't think it is such a big deal.

While I'm not familiar with Android programming, would this help?
http://stackoverflow.com/questions/2703244/android-programming-screen-resolution

Also,
I'm not sure how you render your sprites, but surely it is possible to resize 
them before drawing them to your surface?
```

Original comment by `steve.st...@googlemail.com` on 10 Feb 2011 at 8:07

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 24 Feb 2011 at 10:04
- Added labels: **Milestone-0.6.9**

**Comment by GoogleCodeExporter** [comment]
```
I have modified the scaling code to better take the device DPI into effect. 

We scale all tiles during the "loading resources"-screen. Previously, we 
rescaled all tiles to a measured suggested pixel width depending on the 
automatic device scaling of images.

Now, we have two scaling methods available. The first, pre-scaling during 
"loading resources" is now done according to the device DPI. This scaling will 
use dithering to make the tiles look good even when scaled up. This scaling 
will never change during the lifetime of the application, since the device 
density most likely will not change during the aplication lifetime. Therefore, 
we can do this expensive scaling on startup.

Secondly, I have introduced a preference setting to allow the player to scale 
up the tiles additionally, if the device DPI does not completely satisfy the 
scaling needs. This could be the case, for example, on a device with large 
resolution (1024x600) but low dpi (160), where the pre-scaling would not 
sufficiently scale the tiles. With this setting, the player can scale all tiles 
when they are drawn instead. Since this scaling is done on the fly when 
drawing, it must be super-quick and not use any fancy methods such as 
dithering. Therefore, the scaling done by the preference setting will look kind 
of pixel:y and blocky.

I am marking this issue as ready for the next release.

I hope this will work out well for the players using tablets!
```

Original comment by `oskar.wi...@gmail.com` on 9 Mar 2011 at 7:36
- Changed state: **Ready**

**Comment by GoogleCodeExporter** [comment]
```
I should also add that we will now support all displays on any type of device, 
including future device sizes that are not yet specified. The game will now 
scale automatically to them.
```

Original comment by `oskar.wi...@gmail.com` on 9 Mar 2011 at 7:40

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.9 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 29 Jun 2011 at 7:49
- Changed state: **Done**
